lib/aws/client.ex

```diff
@@ -195,11 +195,61 @@ defmodule AWS.Client do
   end
 
   def request(client, method, url, body, headers, opts \\ []) do
```

Comment:
Maybe we can add a small doc explaining the `retry_opts` keyword and what it may contain? 

Something along the lines of (feel free to rephrase/rework):
```elixir
@doc """
`opts` may contain the keywords `:retry_num` for how often retries should be attempted as well as 
the keyword `:retry_opts` which in and of itself is a keyword list consisting of:
`:max_retries`, `:base_sleep_time` & `:cap_sleep_time` 

This uses exponential backoff with jitter to sleep in between retry attempts. 

See: https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
"""
```